### PR TITLE
pulumi 0.15.0 (new formula)

### PR DESCRIPTION
**Pull request by tschersach**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Review by commitay — DISMISSED** [review]
This would need to build from source to be accepted into `homebrew/core`

https://docs.brew.sh/Acceptable-Formulae

**Comment by Tirke** [comment]
Thanks for the quick feedback, I will update the formula to build from source.

File: Formula/pulumi.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Pulumi < Formula
+  desc "Define cloud applications and infrastructure in your favorite language"
+  homepage "https://pulumi.io/"
+  url "https://github.com/pulumi/pulumi.git",
+      :tag => "v0.14.3"
```
Comment (DomT4) [inline Formula/pulumi.rb]: This needs to have a `revision` as well (the commit the tag represents).

File: Formula/pulumi.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+  depends_on "node@8" => :build
+  depends_on "pandoc" => :build
+  depends_on "pipenv" => :build
```
Comment (DomT4) [inline Formula/pulumi.rb]: Are we sure absolutely all of these are `buildtime` only?

File: Formula/pulumi.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "python@2" => :build
+  depends_on "yarn" => :build
+
+  def install
+    ENV.deparallelize
```
Comment (DomT4) [inline Formula/pulumi.rb]: This is regarded as a bug with new formulae & should be reported upstream as such. Parallel builds shouldn't be problematic.

File: Formula/pulumi.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    ENV["GOPATH"] = buildpath
+    # Remove once opt/pulumi/bin is not needed anymore
+    ENV["PULUMI_ROOT"] = buildpath / "opt/pulumi/bin"
```
Comment (DomT4) [inline Formula/pulumi.rb]: Please cull the excessive spacing. `buildpath/"opt/pulumi/bin"` and so on will work just fine.

File: Formula/pulumi.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    cd "src/github.com/pulumi/pulumi" do
+      system "make", "ensure"
+      system "make", "only_build"
+      bin.install Dir["#{buildpath}/bin/*"]
+    end
+  end
+
+  test do
+    output = shell_output("#{bin}/pulumi version")
```
Comment (DomT4) [inline Formula/pulumi.rb]: Please try and add a test that tests functionality more comprehensively.

**Review by DomT4 — CHANGES_REQUESTED** [review]

**Comment by Tirke** [comment]
@DomT4 Thanks for all the great feedback, I have made numerous changes to the formula and all the problems you pointed should be fixed. I'm just waiting on upstream PRs to be merged to push my changes to the formula.

**Comment by DomT4** [comment]
@Tirke No worries! I think our inactivity bot gives you about 3 weeks after the last comment to make changes before it gets nervous, so you've got plenty of time 😄.

**Comment by Tirke** [comment]
Formula updated, everything should be fine now.

File: Formula/pulumi.rb [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      :revision => "23cbfa503d856f57a3e00599d53c04fe69c93cf0"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    dir = buildpath/"src/github.com/pulumi/pulumi"
+    dir.install buildpath.children
+    (buildpath/"bin").mkpath
```
Comment (DomT4) [inline Formula/pulumi.rb]: What's happening here to necessitate this?

File: Formula/pulumi.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+    cd dir do
+      system "dep", "ensure", "-vendor-only"
+      system "make", "dist"
+      bin.install Dir["#{buildpath}/bin/*"]
```
Comment (DomT4) [inline Formula/pulumi.rb]: You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

**Comment by DomT4** [comment]
> Edit : I'm also wondering when does the bottle part of a formula gets added ? Is this done on a separate PR, automagically or on the first version bump ?

automagically, essentially.

File: Formula/pulumi.rb (hunk 6 above)
Comment (Tirke) [reply Formula/pulumi.rb]: It's a pretty standard set of operations for a `go`-using formula ? I don't really get what is bothering you sorry 😞

File: Formula/pulumi.rb (hunk 7 above)
Comment (Tirke) [reply Formula/pulumi.rb]: I've added it but I don't really understand why I need things like README and LICENSE into the prefix.

**Comment by Tirke** [comment]
I hope that everything is ok now 😄

File: Formula/pulumi.rb (hunk 7 above)
Comment (DomT4) [reply Formula/pulumi.rb]: IANAL but I think most open-source licenses _at least_ imply an expectation of providing the license when you package that software in source or binary format. The formula `prefix` just happens to be where we put those.

File: Formula/pulumi.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
+class Pulumi < Formula
+  desc "Define cloud applications and infrastructure in your favorite language"
```
Comment (DomT4) [inline Formula/pulumi.rb]: I think this sounds a little... sales pitchy. The description on GitHub is maybe worth considering instead:
```
Cloud native development platform
```
or:
```
Cloud native language-agnostic development platform
```
or something similar.

**Review by DomT4 — CHANGES_REQUESTED** [review]
Almost almost almost there.

File: Formula/pulumi.rb (hunk 7 above)
Comment (Tirke) [reply Formula/pulumi.rb]: Makes sense, thanks.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/a06a6aab3238d4b0eb2e49a217fa237864f23b80. Thank you for your contribution to Homebrew @Tirke; we appreciate it! :smiley_cat:

**Pull request closed without merging**